internal/browser/browser.go

```diff
@@ -49,6 +50,9 @@ func OpenBrowserCommand(runner backendRunner) Option[string] {
 			return Some(browserCommand)
 		}
 	}
+	if browser := os.Getenv("BROWSER"); browser != "" {
```

Comment:
If there is an explicit environment variable specifying a browser, why do we use it as a fallback? Would it make more sense to treat it as `EDITOR` or `GIT_EDITOR` and use try it first?

Reply:
I guess if the user has specified a browser, we should use it. I wanted to add this in a way that wouldn't change existing behavior as much as possible so I used a fallback.

Reply:
@kevgo WDYT?

Reply:
Let's implement this so that the new behavior is the most intuitive to our users. Not changing existing behavior unnecessarily is important, but following the [principle of least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) is even more important in my opinion.

If I define a `BROWSER` environment variable to tell applications which of the installed browsers to use, I would expect that an application like Git Town uses this override from the environment, no matter what other browsers are installed.

I view this PR as adding additional behavior, rather than changing existing behavior.

With that in mind, I agree with @legeana on this one.